### ISSUE TITLE
Support start/end date and max submissions for forms

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#2432]
This calls for a form published to be automatically be made available/unavailable depending on rules defined by the form author:

- In some cases, form authors would like to publish a form, but have that form available only after a certain "start" date, and not be available after another "end" date.
- In some cases, form authors don't want to have more than a certain number of submissions of the form, and want the form to automatically become unavailable after that number has been reached.

**Comment by ebruchez** [comment]
Could this be implemented on top of workflow? This would require the ability to change a form's stage asynchronously, based on a particular date/time being reached.

**Comment by avernet** [comment]
Yes, good thinking. We were planning, as some point, to allow the stage to change based on time. In addition, we would need to allow the stage to change based on the number of submissions, but I think this is the right way to go.

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/4224335/documents/3933134103)

**Comment by avernet** [comment]
Some thoughts about potential changes we might want to do to the Form Runner Forms page:

- Tentative requirement
  - (1) If the form isn't available yet, but there is a previous version that is, we should link to that version
  - (2) If the form isn't available anymore, but users have data they have access, the form should still show
    - We have this now, for users who can't create data, but could access to data created by others
- Problematic case: user can read their own data, v2 isn't available yet, should we link to v1 or v2?
  - Per (1), we should link to v1/new, as v2 isn't available yet
  - Per (2), we should link to v2/summary, as they could access data in v2 if there are any
  - Intuitively, I'd like to say: v2/summary if data that page wouldn't be empty, v1/new otherwise
    - But that would require to check if there are data, which is another request, per app/name/version, which we don't do for now in the metadata API (and for a good reason)
- Alternative: ignore availability, at least for now
